src/extension/prompts/node/panel/promptFile.tsx

```diff
@@ -62,8 +65,8 @@ export class PromptFile extends PromptElement<PromptFileProps, void> {
 
 	private async getBodyContent(fileUri: URI, toolReferences: readonly ChatLanguageModelToolReference[] | undefined): Promise<string | undefined> {
 		try {
-			const fileContents = await this.fileSystemService.readFile(fileUri);
-			let content = new TextDecoder().decode(fileContents);
+			const documentText = this.workspaceService.textDocuments.find(doc => isEqual(doc.uri, fileUri))?.getText();
+			let content = documentText ?? new TextDecoder().decode(await this.fileSystemService.readFile(fileUri));
```

Comment:
I suppose this change would also work in the rare case we have uncommitted changes to the prompt file.